### PR TITLE
Update stakeshuffle page UI

### LINKED ISSUE #1019: v2 stake shuffle layout update.
update the stake shuffle layout to the v2 layout.

image

https://www.figma.com/file/e0cYaAzkXyXXPY00v23cQY/GoDCR-%26-DCRMobile?node-id=3307%3A128667

### LINKED ISSUE #1040: App crash when stakeshufle page is navigated to
Godcr crashes when the stakeshuffle page is navigated to

### PR DESCRIPTION
**Pull request by devchoplife**
Closes #1019 
Fixes #1040 
This PR updates the mixer page to the latest UI

image

### CODE REVIEW

File: ui/page/privacy/account_mixer_page.go [hunk 1]
```diff
@@ -33,27 +32,28 @@ type AccountMixerPage struct {
...
+		Load:             l,
+		GenericPageModal: app.NewGenericPageModal(AccountMixerPageID),
+		wallet:           l.WL.SelectedWallet.Wallet,
+		toggleMixer:      l.Theme.Switch(),
+		mixerProgress:    l.Theme.ProgressBar(0),
 	}
-	pg.backButton, pg.infoButton = components.SubpageHeaderButtons(l)
+	pg.mixerProgress.Height = values.MarginPadding18
+	pg.mixerProgress.Radius = decredmaterial.Radius(2)
+	totalBalance, _ := components.CalculateTotalWalletsBalance(pg.Load)
```
Comment (JustinBeBoy) [inline ui/page/privacy/account_mixer_page.go]: We need to handle error if occur

File: ui/page/privacy/account_mixer_page.go [hunk 2]
```diff
@@ -68,175 +68,176 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
 	pg.listenForMixerNotifications()
 	pg.toggleMixer.SetChecked(pg.WL.SelectedWallet.Wallet.IsAccountMixerActive())
 
-	isSpendUnmixedFunds := pg.WL.SelectedWallet.Wallet.ReadBoolConfigValueForKey(load.SpendUnmixedFundsKey, false)
-	pg.allowUnspendUnmixedAcct.SetChecked(isSpendUnmixedFunds)
+	/*
+		isSpendUnmixedFunds := pg.wallet.ReadBoolConfigValueForKey(load.SpendUnmixedFundsKey, false)
```
Comment (JustinBeBoy) [inline ui/page/privacy/account_mixer_page.go]: please remove it if not necessary

File: ui/page/privacy/account_mixer_page.go [hunk 3]
```diff
@@ -68,175 +68,176 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
-										}),
-										layout.Rigid(func(gtx C) D {
-											return layout.Flex{Spacing: layout.SpaceBetween, Alignment: layout.Middle}.Layout(gtx,
-												layout.Rigid(func(gtx C) D {
-													t := pg.Theme.Label(values.TextSize14, "Mixed balance")
-													t.Color = pg.Theme.Color.GrayText2
-													return t.Layout(gtx)
-												}),
+func (pg *AccountMixerPage) MixerInfoLayout(gtx C, l *load.Load, mixerActive bool, button layout.Widget) D {
+	mixedBalance := "0.00"
```
Comment (JustinBeBoy) [inline ui/page/privacy/account_mixer_page.go]: Just 0, Need to agree on how to display the balance

File: ui/page/privacy/account_mixer_page.go [hunk 4]
```diff
@@ -68,175 +68,176 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
+	unmixedBalance := "0.00"
+	accounts, _ := pg.wallet.GetAccountsRaw()
```
Comment (JustinBeBoy) [inline ui/page/privacy/account_mixer_page.go]: need handles the error if occur

File: ui/page/privacy/account_mixer_page.go [hunk 5]
```diff
@@ -68,175 +68,176 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
+											return components.LayoutBalance(gtx, pg.Load, pg.totalBalance.String())
+										})
+									}),
+									layout.Flexed(1, func(gtx C) D {
+										return layout.E.Layout(gtx, func(gtx C) D {
+											txt := l.Theme.H6(values.String(values.StrMix))
+											return layout.Flex{Alignment: layout.Middle}.Layout(gtx,
 												layout.Rigid(func(gtx C) D {
-													return components.LayoutBalance(gtx, pg.Load, mixedBalance)
+													return layout.Inset{Right: values.MarginPadding10}.Layout(gtx, func(gtx C) D {
```
Comment (JustinBeBoy) [inline ui/page/privacy/account_mixer_page.go]: Should be : `return layout.Inset{Right: values.MarginPadding10}.Layout(gtx, txt.Layout)`

File: ui/page/privacy/account_mixer_page.go [hunk 6]
```diff
@@ -68,175 +68,176 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
+						layout.Rigid(func(gtx C) D {
+							return layout.Inset{Top: values.MarginPadding15}.Layout(gtx, func(gtx C) D {
+								return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Middle}.Layout(gtx,
+									layout.Rigid(func(gtx C) D {
+										txt := l.Theme.H6(values.String(values.StrMixer))
+										return txt.Layout(gtx)
+									}),
+									layout.Rigid(func(gtx C) D {
+										return layout.Inset{Left: values.MarginPadding20, Right: values.MarginPadding40}.Layout(gtx, func(gtx C) D {
+											return pg.mixerProgress.Layout(gtx)
```
Comment (JustinBeBoy) [inline ui/page/privacy/account_mixer_page.go]: Should be `return layout.Inset{Left: values.MarginPadding20, Right: values.MarginPadding40}.Layout(gtx, pg.mixerProgress.Layout)`

File: ui/page/privacy/account_mixer_page.go [hunk 7]
```diff
@@ -68,175 +68,176 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
+			layout.Rigid(func(gtx C) D {
+				return layout.Inset{Right: values.MarginPadding15, Left: values.MarginPadding15, Top: values.MarginPadding15}.Layout(gtx, func(gtx C) D {
+					return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Middle}.Layout(gtx,
+						layout.Rigid(func(gtx C) D {
+							ic := l.Theme.Icons.MixedTxIcon
+							return ic.Layout12dp(gtx)
+						}),
+						layout.Rigid(func(gtx C) D {
+							txt := l.Theme.H6(values.String(values.StrMixed))
+							return layout.Inset{Left: values.MarginPadding11}.Layout(gtx, func(gtx C) D {
```
Comment (JustinBeBoy) [inline ui/page/privacy/account_mixer_page.go]: Should be `return layout.Inset{Left: values.MarginPadding11}.Layout(gtx, txt.Layout)`

File: ui/page/privacy/account_mixer_page.go [hunk 8]
```diff
@@ -68,175 +68,176 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
+				return layout.Inset{Right: values.MarginPadding40, Left: values.MarginPadding40}.Layout(gtx, func(gtx C) D {
+					return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Middle}.Layout(gtx,
+						layout.Rigid(func(gtx C) D {
+							gtx.Constraints.Max.X = gtx.Constraints.Max.X/2 - 20
+							return layout.W.Layout(gtx, l.Theme.Separator().Layout)
+						}),
+						layout.Rigid(func(gtx C) D {
+							return layout.Inset{Left: values.MarginPadding10, Right: values.MarginPadding10}.Layout(gtx, func(gtx C) D {
+								ic := l.Theme.Icons.MixerIcon
+								return layout.Center.Layout(gtx, func(gtx C) D {
```
Comment (JustinBeBoy) [inline ui/page/privacy/account_mixer_page.go]: Should be return layout.Center.Layout(gtx, ic.Layout36dp)

File: ui/page/privacy/account_mixer_page.go [hunk 9]
```diff
@@ -68,175 +68,176 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
+						}),
+					)
+				})
+			}),
+			layout.Rigid(func(gtx C) D {
+				return layout.Inset{Right: values.MarginPadding15, Left: values.MarginPadding15, Bottom: values.MarginPadding40}.Layout(gtx, func(gtx C) D {
+					return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Middle}.Layout(gtx,
+						layout.Rigid(func(gtx C) D {
+							ic := l.Theme.Icons.UnmixedTxIcon
+							return ic.Layout12dp(gtx)
```
Comment (JustinBeBoy) [inline ui/page/privacy/account_mixer_page.go]: the same line 174

File: ui/page/privacy/account_mixer_page.go [hunk 10]
```diff
@@ -268,41 +269,39 @@ func (pg *AccountMixerPage) HandleUserInteractions() {
...
-						tim.SetError("confirmation text is incorrect")
-						tim.SetLoading(false)
-					} else {
-						pg.WL.SelectedWallet.Wallet.SetBoolConfigValueForKey(load.SpendUnmixedFundsKey, true)
-						tim.Dismiss()
-					}
-					return false
-				})
+	/*
+		if pg.allowUnspendUnmixedAcct.Changed() {
```
Comment (JustinBeBoy) [inline ui/page/privacy/account_mixer_page.go]: please remove it if not necessary

File: ui/page/privacy/account_mixer_page.go [hunk 11]
```diff
@@ -268,41 +269,39 @@ func (pg *AccountMixerPage) HandleUserInteractions() {
...
+						}
+						return false
+					})
 
-			textModal.Title("Confirm to allow spending from unmixed accounts").
-				NegativeButton("Cancel", func() {
-					pg.allowUnspendUnmixedAcct.SetChecked(false)
-				})
-			pg.ParentWindow().ShowModal(textModal)
+				textModal.Title("Confirm to allow spending from unmixed accounts").
```
Comment (JustinBeBoy) [inline ui/page/privacy/account_mixer_page.go]: please remove it if not necessary

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]
I can't start mixing
image

**Comment by devchoplife** [comment]
> I can't start mixing image

This PR focuses on the UI
The functionality of the mixer is excluded

File: ui/page/privacy/account_mixer_page.go (hunk 3 above)
Comment (devchoplife) [reply ui/page/privacy/account_mixer_page.go]: As I said earlier, this PR focuses on the UI and not functionality

**Comment by JustinBeBoy** [comment]
> > I can't start mixing image
> 
> This PR focuses on the UI The functionality of the mixer is excluded

I can't start mixing at the moment so that hinders the review of this PR, I can't see the way of the progress works

File: ui/page/privacy/account_mixer_page.go (hunk 9 above)
Comment (devchoplife) [reply ui/page/privacy/account_mixer_page.go]: This does not apply because the function expects layout.Dimension

**Comment by devchoplife** [comment]
> > > I can't start mixing image
> > 
> > 
> > This PR focuses on the UI The functionality of the mixer is excluded
> 
> I can't start mixing at the moment so that hinders the review of this PR, I can't see the way of the progress works

The progress bar functionality has not been implemented, this PR only updates the UI to V2

File: ui/page/components/acct_mixer.go [hunk 12]
```diff
@@ -1,13 +1,7 @@
...
-import (
-	"gioui.org/layout"
-
-	"github.com/planetdecred/godcr/ui/load"
-	"github.com/planetdecred/godcr/ui/values"
-)
-
-func mixerInfoStatusTextLayout(gtx C, l *load.Load, mixerActive bool) D {
+/*
+func MixerInfoStatusTextLayout(gtx C, l *load.Load, mixerActive bool) D {
```
Comment (Sirmorrison) [inline ui/page/components/acct_mixer.go]: Remove if no longer needed.

File: ui/page/components/acct_mixer.go [hunk 13]
```diff
@@ -81,3 +40,4 @@ func MixerInfoContentWrapper(gtx C, l *load.Load, content layout.Widget) D {
 		return layout.UniformInset(values.MarginPadding15).Layout(gtx, content)
 	})
 }
+*/
```
Comment (Sirmorrison) [inline ui/page/components/acct_mixer.go]: remove is not used

File: ui/page/main_page.go [hunk 14]
```diff
@@ -748,7 +748,7 @@ func (mp *MainPage) LayoutUSDBalance(gtx layout.Context) layout.Dimensions {
 	}
 }
 
-func (mp *MainPage) totalDCRBalance(gtx C) D {
+func (mp *MainPage) TotalDCRBalance(gtx C) D {
```
Comment (Sirmorrison) [inline ui/page/main_page.go]: Why do we have this as a global variable when it is only used on this page? Why was this change made? Kindly revert if it does not impact your PR.

File: ui/page/main_page.go [hunk 15]
```diff
@@ -819,7 +819,7 @@ func (mp *MainPage) LayoutTopBar(gtx layout.Context) layout.Dimensions {
 					return layout.E.Layout(gtx, func(gtx C) D {
 						return layout.Flex{}.Layout(gtx,
 							layout.Rigid(func(gtx C) D {
-								return mp.totalDCRBalance(gtx)
+								return mp.TotalDCRBalance(gtx)
```
Comment (Sirmorrison) [inline ui/page/main_page.go]: revert.

File: ui/page/privacy/account_mixer_page.go [hunk 16]
```diff
@@ -67,177 +67,155 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
-										}),
-									)
+										})
+									}),
+								)
+							})
+						}),
+						layout.Rigid(func(gtx C) D {
+							return l.Theme.Separator().Layout(gtx)
+						}),
```
Comment (Sirmorrison) [inline ui/page/privacy/account_mixer_page.go]:
```suggestion
						}),
						layout.Rigid( l.Theme.Separator().Layout),
```

File: ui/page/privacy/account_mixer_page.go [hunk 17]
```diff
@@ -67,177 +67,155 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
+					)
+				})
+			}),
+			layout.Rigid(func(gtx C) D {
+				return layout.Inset{Right: values.MarginPadding15, Left: values.MarginPadding15, Top: values.MarginPadding15}.Layout(gtx, func(gtx C) D {
+					return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Middle}.Layout(gtx,
+						layout.Rigid(func(gtx C) D {
+							ic := l.Theme.Icons.MixedTxIcon
+							return ic.Layout12dp(gtx)
```
Comment (Sirmorrison) [inline ui/page/privacy/account_mixer_page.go]:
```suggestion
							return  l.Theme.Icons.MixedTxIcon.Layout12dp(gtx)
```

File: ui/page/privacy/account_mixer_page.go [hunk 18]
```diff
@@ -67,177 +67,155 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
+						}),
+						layout.Rigid(func(gtx C) D {
+							return l.Theme.Separator().Layout(gtx)
+						}),
+						layout.Rigid(func(gtx C) D {
+							return layout.Inset{Top: values.MarginPadding15}.Layout(gtx, func(gtx C) D {
+								return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Middle}.Layout(gtx,
+									layout.Rigid(func(gtx C) D {
+										txt := l.Theme.H6(values.String(values.StrMixer))
+										return txt.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/privacy/account_mixer_page.go]:
```suggestion
										return  l.Theme.H6(values.String(values.StrMixer)).Layout(gtx)
```

File: ui/page/privacy/account_mixer_page.go [hunk 19]
```diff
@@ -67,177 +67,155 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
+					return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Middle}.Layout(gtx,
+						layout.Rigid(func(gtx C) D {
+							ic := l.Theme.Icons.UnmixedTxIcon
+							return ic.Layout12dp(gtx)
+						}),
+						layout.Rigid(func(gtx C) D {
+							txt := l.Theme.H6(values.String(values.StrUnmixed))
+							return layout.Inset{Left: values.MarginPadding11}.Layout(gtx, func(gtx C) D {
+								return txt.Layout(gtx)
+							})
```
Comment (Sirmorrison) [inline ui/page/privacy/account_mixer_page.go]:
```suggestion
							return layout.Inset{Left: values.MarginPadding11}.Layout(gtx, txt.Layout)
```

File: ui/page/privacy/account_mixer_page.go [hunk 20]
```diff
@@ -67,177 +67,155 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
+					)
+				})
+			}),
+			layout.Rigid(func(gtx C) D {
+				return layout.Inset{Right: values.MarginPadding15, Left: values.MarginPadding15, Bottom: values.MarginPadding40}.Layout(gtx, func(gtx C) D {
+					return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Middle}.Layout(gtx,
+						layout.Rigid(func(gtx C) D {
+							ic := l.Theme.Icons.UnmixedTxIcon
+							return ic.Layout12dp(gtx)
+						}),
```
Comment (Sirmorrison) [inline ui/page/privacy/account_mixer_page.go]:
```suggestion
						layout.Rigid(layout.Rigid(l.Theme.Icons.UnmixedTxIcon.Layout12dp),
```

File: ui/page/privacy/account_mixer_page.go [hunk 21]
```diff
@@ -67,177 +67,155 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
+					return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Middle}.Layout(gtx,
+						layout.Rigid(func(gtx C) D {
+							gtx.Constraints.Max.X = gtx.Constraints.Max.X/2 - 20
+							return layout.W.Layout(gtx, l.Theme.Separator().Layout)
+						}),
+						layout.Rigid(func(gtx C) D {
+							return layout.Inset{Left: values.MarginPadding10, Right: values.MarginPadding10}.Layout(gtx, func(gtx C) D {
+								ic := l.Theme.Icons.MixerIcon
+								return layout.Center.Layout(gtx, ic.Layout36dp)
+							})
```
Comment (Sirmorrison) [inline ui/page/privacy/account_mixer_page.go]:
```suggestion
							return layout.Inset{Left: values.MarginPadding10, Right: values.MarginPadding10}.Layout(gtx, func(gtx C) D {
								return layout.Center.Layout(gtx, l.Theme.Icons.MixerIcon.Layout36dp)
							})
```

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
image
The icon seems blurry and not centralized

**Comment by devchoplife** [comment]
> image
> 
> The icon seems blurry and not centralized

I extracted from figma
I will check if I can get a clearer one

File: ui/page/privacy/account_mixer_page.go [hunk 22]
```diff
@@ -67,177 +67,143 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
+								)
+							})
+						}),
+						layout.Rigid(l.Theme.Separator().Layout),
+						layout.Rigid(func(gtx C) D {
+							return layout.Inset{Top: values.MarginPadding15}.Layout(gtx, func(gtx C) D {
+								return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Middle}.Layout(gtx,
+									layout.Rigid(func(gtx C) D {
+										return l.Theme.H6(values.String(values.StrMixer)).Layout(gtx)
+									}),
```
Comment (Sirmorrison) [inline ui/page/privacy/account_mixer_page.go]:
```suggestion
									layout.Rigid(l.Theme.H6(values.String(values.StrMixer)).Layout),
```

File: ui/page/privacy/account_mixer_page.go [hunk 23]
```diff
@@ -67,177 +67,143 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
+									layout.Rigid(func(gtx C) D {
+										return layout.Inset{Left: values.MarginPadding20, Right: values.MarginPadding40}.Layout(gtx, pg.mixerProgress.Layout)
```
Comment (Sirmorrison) [inline ui/page/privacy/account_mixer_page.go]:
```suggestion
										return layout.Inset{
										  Left: values.MarginPadding20, 
										  Right: values.MarginPadding40,
										}.Layout(gtx, pg.mixerProgress.Layout)
```

File: ui/page/privacy/account_mixer_page.go [hunk 24]
```diff
@@ -67,177 +67,143 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
+									}),
+								)
+							})
+						}),
+					)
+				})
+			}),
+			layout.Rigid(func(gtx C) D {
+				return layout.Inset{Right: values.MarginPadding15, Left: values.MarginPadding15, Top: values.MarginPadding15}.Layout(gtx, func(gtx C) D {
```
Comment (Sirmorrison) [inline ui/page/privacy/account_mixer_page.go]:
```suggestion
                              insetValue := values.MarginPadding15
				return layout.Inset{
				  Right: insetValue, 
				  Left: insetValue, 
				  Top: insetValue,
				}.Layout(gtx, func(gtx C) D {
```

File: ui/page/privacy/account_mixer_page.go [hunk 25]
```diff
@@ -67,177 +67,143 @@ func (pg *AccountMixerPage) OnNavigatedTo() {
...
+							})
+						}),
+						layout.Rigid(func(gtx C) D {
+							return layout.E.Layout(gtx, l.Theme.Separator().Layout)
+						}),
+					)
+				})
+			}),
+			layout.Rigid(func(gtx C) D {
+				return layout.Inset{Right: values.MarginPadding15, Left: values.MarginPadding15, Bottom: values.MarginPadding40}.Layout(gtx, func(gtx C) D {
```
Comment (Sirmorrison) [inline ui/page/privacy/account_mixer_page.go]:
```suggestion
				return layout.Inset{
				  Right: values.MarginPadding15, 
				  Left: values.MarginPadding15, 
				  Bottom: values.MarginPadding40,
				}.Layout(gtx, func(gtx C) D {
```

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

File: ui/page/main_page.go (hunk 14 above)
Comment (JustinBeBoy) [reply ui/page/main_page.go]: @devchoplife  please revert it

**Comment by JustinBeBoy** [comment]
@devchoplife  I see many comments @Sirmorrison that you marked solved, I checked again but it is not resolved, please check it again

**Comment by dreacot** [comment]
> > I can't start mixing image
> 
> This PR focuses on the UI The functionality of the mixer is excluded

Why does this PR just focus on UI, the more UI we implement without functionality, the farther it takes us away from out goal.
Excepts there's a block in implementing the functionality, let's resolve it at the same time.

Because there isn't even a proper way to ensure the UI implementation is correct if the functionality doesn't work

File: ui/page/components/acct_mixer.go [hunk 26]
```diff
@@ -38,41 +32,6 @@ func mixerInfoStatusTextLayout(gtx C, l *load.Load, mixerActive bool) D {
...
-					return D{}
-				}),
-			)
-		})
-	})
-}
-
 func MixerInfoContentWrapper(gtx C, l *load.Load, content layout.Widget) D {
 	card := l.Theme.Card()
 	card.Color = l.Theme.Color.Gray4
```
Comment (dreacot) [inline ui/page/components/acct_mixer.go]: let's remove the comments, are they needed?

**Review by dreacot — CHANGES_REQUESTED** [review]
seems like the layout can still be improved;

1. The trailing "DCR" of the balance should be bolder and larger
2. the mixed and unmixed balances should have a trailing DCR, the decimals can go for zero balances as well
3. the "Mixer" text should be a lighter shade of gray
4.  looks like the max width of the mixer card should be reduced

Screenshot from 2022-08-03 11-17-43

Screenshot from 2022-08-03 11-17-56

**Pull request closed without merging**